### PR TITLE
Promote existing E2Es for SchedulerPreemption to conformance

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR is a request for promotion of the existing E2Es to conformance:
1. "validates basic preemption works"
2. "validates lower priority pod preemption by critical pod"
3. "validates pod anti-affinity works in preemption"

**Which issue(s) this PR fixes**: NONE - Promotion of existing E2Es is requested via this PR
Fixes #

**Special notes for your reviewer**:
1. "validates basic preemption works":
         --- Takes approximately 92 seconds for successful execution.
         --- The test verifies that the basic preemption logic is validated i.e. when a lower priority pod is 
         removed to schedule a higher priority pod when no nodes with the necessary resources can be 
         found to schedule the higher priority pod.

2. "validates lower priority pod preemption by critical pod":
         --- Takes approximately 80 seconds for successful execution.
         --- The test validates that the lower priority pod is vacated when no nodes are available to schedule
         a critical pod.

3. "validates pod anti-affinity works in preemption":
         --- Takes approximately 88 seconds for successful execution.
         --- The test validates that the anti-affinity rules are taken into account during pod preemption.

All three existing tests are found to be on-disruptive and non-flaky in nature.

NOTE: As per Kubernetes API document for v1.14, PriorityClass and Affinity are in v1.

**Does this PR introduce a user-facing change?**: NONE

/release-note-none

/area conformance

cc @mgdevstack @brahmaroutu

### CODE REVIEW

File: test/e2e/scheduling/preemption.go [hunk 1]
```diff
@@ -89,7 +89,13 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 	// This test verifies that when a higher priority pod is created and no node with
 	// enough resources is found, scheduler preempts a lower priority pod to schedule
 	// the high priority pod.
-	ginkgo.It("validates basic preemption works", func() {
+	/*
+		Release : v1.15
```
Comment (johnbelamaric) [inline test/e2e/scheduling/preemption.go]: 1.16 now

"The test SHOULD pass" is not a useful way to word these. The MUST/SHOULD etc. needs to be about the behavior, not about the test. How about this instead:

```
Description: When a higher priority pod is created and no node with enough resources is found, scheduler MUST preempt a lower priority pod and schedule the high priority pod.
```

File: test/e2e/scheduling/preemption.go [hunk 2]
```diff
@@ -149,7 +155,14 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 	// This test verifies that when a critical pod is created and no node with
 	// enough resources is found, scheduler preempts a lower priority pod to schedule
 	// this critical pod.
-	ginkgo.It("validates lower priority pod preemption by critical pod", func() {
+	/*
+		Release : v1.15
```
Comment (johnbelamaric) [inline test/e2e/scheduling/preemption.go]: Ditto 1.16 and

```
Description: When a critical pod is created and no node with enough resources is found, the scheduler MUST preempt a lower priority pod to schedule the critical pod.
```

File: test/e2e/scheduling/preemption.go [hunk 3]
```diff
@@ -217,7 +230,14 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 	// pod is preempted to allow the higher priority pod schedule.
 	// It also verifies that existing low priority pods are not preempted as their
 	// preemption wouldn't help.
-	ginkgo.It("validates pod anti-affinity works in preemption", func() {
+	/*
+		Release : v1.15
```
Comment (johnbelamaric) [inline test/e2e/scheduling/preemption.go]: 1.16
Again description should describe the desired behavior.

```
Description: When a high priority pod is pending and its scheduling violates a medium priority pod anti-affinity,
the medium priority pod MUST be preempted to allow the higher priority pod schedule.
Existing low priority pods MUST NOT be preempted as their preemption wouldn't help.
```

**Review by johnbelamaric — CHANGES_REQUESTED** [review]
I reviewed the metadata but still haven't looked into the tests themselves to make sure they are working as intended.

**Comment by bclau** [comment]
/retest

FWIW, the tests pass on Windows.

**Comment by johnbelamaric** [comment]
@bsalamat can you confirm that these tests are correct? If you can confirm, then I am good with promoting them.

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Comment by johnbelamaric** [comment]
@globervinodhn So, functionally these seem to be OK, but somehow this v1 -> corev1 change got put in there and it is failing to build.

File: test/e2e/scheduling/preemption.go [hunk 4]
```diff
@@ -91,8 +91,14 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
...
-	ginkgo.It("validates basic preemption works", func() {
-		var podRes v1.ResourceList
+	/*
+		Release : v1.16
+		Testname: Scheduler, Basic Preemption
+		Description: When a higher priority pod is created and no node with enough resources is found,
+		scheduler MUST preempt a lower priority pod and schedule the high priority pod.
+	*/
+	framework.ConformanceIt("validates basic preemption works", func() {
+		var podRes corev1.ResourceList
```
Comment (johnbelamaric) [inline test/e2e/scheduling/preemption.go]: what is this?

**Comment by spiffxp** [comment]
/ok-to-test
needs rebase

**Comment by spiffxp** [comment]
I'm kicking this back to in-progress since @johnbelamaric's comments haven't been addressed

**Comment by spiffxp** [comment]
/close
Closing in favor of https://github.com/kubernetes/kubernetes/pull/80545

**Pull request closed without merging**